modules/core/src/nan_mask.simd.hpp

```diff
@@ -0,0 +1,439 @@
+            vres = v_extract<1>(vres, vres);
+        }
+    }
+
+    return i;
+}
+
+
+template <>
+int finiteMaskSIMD_<double, 3>(const double *fsrc, uchar *dst, size_t utotal)
+{
+    const uint64_t* src = (const uint64_t*)fsrc;
+    const int npixels = VTraits<v_uint64>::vlanes();
+    v_uint64 vmaskExp = vx_setall_u64(0x7ff0000000000000);
+    v_uint64 z = vx_setzero_u64();
+
+    int i = 0;
+    int total = (int)utotal;
+    for (; i < total - npixels + 1; i += npixels)
+    {
+        v_uint64 vv0, vv1, vv2;
+        vv0 = v_ne(v_and(vx_load(src + i*3            ), vmaskExp), vmaskExp);
+        vv1 = v_ne(v_and(vx_load(src + i*3 +   npixels), vmaskExp), vmaskExp);
+        vv2 = v_ne(v_and(vx_load(src + i*3 + 2*npixels), vmaskExp), vmaskExp);
+
+        v_uint8 velems = v_pack_b(vv0, vv1, vv2, z, z, z, z, z);
+
+        // 2nd arg is useless
+        v_uint8 vsh1 = v_extract<1>(velems, velems);
+        v_uint8 vsh2 = v_extract<2>(velems, velems);
+
+        v_uint8 vres3 = v_and(v_and(velems, vsh1), vsh2);
+        for (int j = 0; j < npixels; j++)
+        {
+            dst[i + j] = v_get0(vres3);
+            // 2nd arg is useless
+            vres3 = v_extract<3>(vres3, vres3);
+        }
+    }
+
+    return i;
+}
+
+template <>
+int finiteMaskSIMD_<double, 4>(const double *dsrc, uchar *dst, size_t utotal)
+{
+    const uint64_t* src = (const uint64_t*)dsrc;
+    CV_StaticAssert(VTraits<v_uint8>::max_nlanes*8 <= 1024, "SIMD registers should be not more than 1024 bit wide");
+    // number of pixels in 2 vector registers
+    const int npixels = VTraits<v_uint64>::vlanes()/2;
+
+    v_uint64 vmaskExp = vx_setall_u64(0x7ff0000000000000);
+    v_uint64 z = vx_setzero_u64();
+    v_uint32 vmaskAll4 = vx_setall_u32(0xFFFFFFFF);
+    v_uint32 z32 = vx_setzero_u32();
+
+    int i = 0;
+    int total = (int)utotal;
+    for(; i < total - npixels + 1; i += npixels )
+    {
+        v_uint64 vv0, vv1;
+        vv0 = v_ne(v_and(vx_load(src + i*4            ), vmaskExp), vmaskExp);
+        vv1 = v_ne(v_and(vx_load(src + i*4 + npixels*2), vmaskExp), vmaskExp);
+
+        v_uint8 velems = v_pack_b(vv0, vv1, z, z, z, z, z, z);
+        v_uint32 vresWide = v_eq(v_reinterpret_as_u32(velems), vmaskAll4);
+
+        v_uint8 vres = v_pack_b(vresWide, z32, z32, z32);
+        if (npixels == 1) // 2x128 bit
+        {
+            dst[i] = v_get0(vres);
+        }
+        else if (npixels == 2) // 2x256 bit
+        {
+            *((uint16_t*)(dst + i)) = v_get0(v_reinterpret_as_u16(vres));
+        }
+        else if (npixels == 4) // 2x512 bit
+        {
+            *((uint32_t*)(dst + i)) = v_get0(v_reinterpret_as_u32(vres));
+        }
+        else if (npixels == 8) // 2x1024 bit wide
+        {
+            *((uint64_t*)(dst + i)) = v_get0(v_reinterpret_as_u64(vres));
+        }
+    }
+
+    return i;
+}
+
+#endif
+
+
+template <typename _Tp, int cn>
+void finiteMask_(const uchar *src, uchar *dst, size_t total)
+{
+    size_t i = 0;
```

Comment:
This is externally exposed function (through `getFiniteMaskFunc`).
`CV_INSTRUMENT_REGION()` is required here to inject `vzeroupper`.

Reply:
Fixed